packages/react-components/src/components/kpi/tests/kpi.spec.tsx

```diff
@@ -0,0 +1,226 @@
+      DATA_STREAM.name
+    );
+    expect(screen.getByTestId('kpi-name-and-unit').textContent).toContain(
+      DATA_STREAM.unit
+    );
+    expect(screen.getByTestId('kpi-value').textContent).toContain(
+      `${DATA_STREAM.data[0].y.toFixed(4)}`
+    );
+    expect(screen.getByTestId('kpi-timestamp').textContent).toContain(
+      new Date(DATA_STREAM.data[0].x).toLocaleString()
+    );
+  });
+
+  it('renders default widget in error state', async () => {
+    render(<KPI query={ERROR_QUERY} />);
+    expect(screen.getByTestId('kpi-error-component').textContent).toContain(
+      ERROR_DATA_STREAM.error.msg
+    );
+  });
+
+  it('renders default widget in loading state', async () => {
+    render(<KPI query={LOADING_QUERY} />);
+
+    expect(screen.getByTestId('kpi-name-and-unit').textContent).toContain('-');
+  });
+});
+
+describe('default kpi with thresholds', () => {
+  it('renders a kpi with a side threhsold', () => {
+    render(<KPI query={SUCCESS_QUERY} thresholds={[THRESHOLD]} />);
+    expect(screen.getByTestId('kpi-name-and-unit').textContent).toContain(
+      DATA_STREAM.name
+    );
+    expect(screen.getByTestId('kpi-name-and-unit').textContent).toContain(
+      DATA_STREAM.unit
+    );
+    expect(screen.getByTestId('kpi-value').textContent).toContain(
+      `${DATA_STREAM.data[0].y.toFixed(4)}`
+    );
+    expect(screen.getByTestId('kpi-timestamp').textContent).toContain(
+      new Date(DATA_STREAM.data[0].x).toLocaleString()
+    );
+    expect(screen.getByTestId('kpi-side-threshold')).toBeVisible();
+    expect(screen.getByTestId('kpi-side-threshold')).toHaveStyle(
+      'background-color: #ff0000'
+    );
+  });
+
+  it('renders a kpi with a filled threhsold', () => {
+    render(<KPI query={SUCCESS_QUERY} thresholds={[FILLED_THRESHOLD]} />);
+    expect(screen.getByTestId('kpi-name-and-unit').textContent).toContain(
+      DATA_STREAM.name
+    );
+    expect(screen.getByTestId('kpi-name-and-unit').textContent).toContain(
+      DATA_STREAM.unit
+    );
+    expect(screen.getByTestId('kpi-value').textContent).toContain(
+      `${DATA_STREAM.data[0].y.toFixed(4)}`
+    );
+    expect(screen.getByTestId('kpi-timestamp').textContent).toContain(
+      new Date(DATA_STREAM.data[0].x).toLocaleString()
+    );
+    expect(screen.getByTestId('kpi-base-component')).toHaveStyle(
+      'background-color: #ff0000'
+    );
+  });
+
+  it('does not render the kpi + side threshold in a loading state', () => {
+    render(<KPI query={LOADING_QUERY} thresholds={[THRESHOLD]} />);
+    expect(screen.queryByTestId('kpi-side-threshold')).toBeNull();
+  });
+
+  it('does not render the kpi + side threshold in an error state', () => {
+    render(
+      <KPI query={ERROR_QUERY} viewport={VIEWPORT} thresholds={[THRESHOLD]} />
+    );
+    expect(screen.queryByTestId('kpi-side-threshold')).toBeNull();
+  });
+});
+
+describe('kpi with custom settings', () => {
+  it('renders a kpi without any extra text', () => {
```

Comment:
What about tests where the settings are `true`? Those would be more valuable than the false equivalents.

Reply:
Also include tests are the different permutations. You don't have to do everything, yet you should have tests for:
1. All true.
2. Tests for each option is true and the rest are false.
3. Tests where defining true for two or more has any special behavior.

Reply:
the true settings are the default which is tested by the entire first group of tests
will add some which test the permutations here !